app/adapters/application.js

```diff
@@ -2,16 +2,33 @@ import Ember from 'ember';
 import DS from 'ember-data';
 import ENV from '../config/environment';
 
+// matches query params or trailing slash
+const TRAILING_SLASH_PATTERN = /((^.*?[^\/])\?(.*?)$|^[^\?]+?[^\/]$)/;
+
 export default DS.RESTAdapter.extend(DS.BuildURLMixin, {
+    session: Ember.inject.service(),
     namespace: 'api/v2',
     host: ENV.apiBaseUrl,
     headers: {
         Accept: 'application/json'
     },
     ajax(url, method, hash) {
+        // trailing slash required for POST requests
+        let urlRegex = new RegExp(TRAILING_SLASH_PATTERN);
+        if (url.match(urlRegex)) {
```

Comment:
```javascript
url.replace(/([^\/])(\?|$)/, '$1/$2')
```

Breaking it down:
Match any non `/` character (`[^/]`) that is followed by either a `?` or the end of the string (`\?|$`)
If there is a match, replace the matched character(s) with the non `/` character, the first capture (`$1`), a `/`, and the second capture (`$2`), either a `?` or empty string.